docs/source/tutorials/qlora_finetune.rst

```diff
@@ -31,10 +31,10 @@ What is QLoRA?
 QLoRA builds on top of LoRA to enable further
 memory savings. In LoRA, model parameters can be thought of as existing in two partitions: adapters, which are
 low-rank matrices added to different layers of a neural network, and base model parameters, which are parameters that are part of
-the original model. In vanilla LoRA-style training, both these parameters are held in the same precision (typically fp32 or bf16), and
+the original model. In vanilla LoRA-style training, both these parameters are held in the same precision (typically fp32 or `bf16 <https://en.wikipedia.org/wiki/Bfloat16_floating-point_format#bfloat16_floating-point_format>`_.), and
 therefore activations and intermediate gradients computed are in fp32/bf16.
 
-QLoRA further quantizes the base model parameters into a bespoke 4-bit NormalFloat (NF4) data type, resulting in 4-8x less parameter memory usage while
+QLoRA further quantizes the base model parameters into a bespoke 4-bit NormalFloat (`NF4 <https://www.youtube.com/watch?v=TPcXVJ1VSRI&t=563s>`_) data type, resulting in 4-8x less parameter memory usage while
```

Comment:
was half expecting this youtube link to be a rick roll

Reply:
I would never do that to you. I actually wonder if we should support the official PyTorch talk [here](https://www.youtube.com/watch?v=dQw4w9WgXcQ) instead?

Reply:
I knew that was coming and I still clicked on it